File: FlexTargetAndroid/app/src/main/java/com/flextarget/android/ui/drills/IDPADrillSummaryView.kt
package com.flextarget.android.ui.drills

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Remove
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.window.Dialog
import com.flextarget.android.R
import com.flextarget.android.data.local.entity.DrillSetupEntity
import com.flextarget.android.data.model.DrillRepeatSummary
import com.flextarget.android.data.model.ScoringUtility
import kotlin.math.abs

@Composable
fun IDPADrillSummaryView(
    summaries: List<DrillRepeatSummary>,
    modifier: Modifier = Modifier,
    onViewResult: (DrillRepeatSummary) -> Unit = {},
    onReplay: (DrillRepeatSummary) -> Unit = {}
) {
    var showEditDialog by remember { mutableStateOf(false) }
    var editingSummary by remember { mutableStateOf<DrillRepeatSummary?>(null) }
    
    Box(modifier = modifier.fillMaxSize()) {
        if (summaries.isEmpty()) {
            EmptyStateView()
        } else {
            LazyColumn(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color.Black),
                contentPadding = PaddingValues(16.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                itemsIndexed(summaries) { index, summary ->
                    IDPADrillCard(
                        summary = summary,
                        index = index,
                        onEditZones = {
                            editingSummary = summary
                            showEditDialog = true
                        },
                        onViewResult = { onViewResult(summary) },
                        onReplay = { onReplay(summary) }
                    )
                }
            }
        }
    }

    // Edit Dialog
    if (showEditDialog && editingSummary != null) {
        IDPAZoneEditDialog(
            summary = editingSummary!!,
            onSave = { updatedZones ->
                editingSummary?.idpaZones = updatedZones
                showEditDialog = false
                editingSummary = null
            },
            onCancel = {
                showEditDialog = false
                editingSummary = null
            }
        )
    }
}

@Composable
private fun IDPADrillCard(
    summary: DrillRepeatSummary,
    index: Int,
    modifier: Modifier = Modifier,
    onEditZones: () -> Unit = {},
    onViewResult: () -> Unit = {},
    onReplay: () -> Unit = {}
) {
    val breakdown = ScoringUtility.getIDPAZoneBreakdown(summary.shots)
    val pointsDown = ScoringUtility.calculateIDPAPointsDown(summary.shots)
    val finalTime = ScoringUtility.calculateIDPAFinalTime(summary.totalTime, pointsDown)

    Card(
        modifier = modifier
            .fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = Color(0xFF1A1A1A)),
        elevation = CardDefaults.cardElevation(defaultElevation = 12.dp)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(20.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Header: Repeat number
            Text(
                text = String.format(stringResource(R.string.repeat_number), summary.repeatIndex),
                style = MaterialTheme.typography.titleMedium,
                color = Color.White,
                fontWeight = FontWeight.SemiBold
            )

            // Metrics row: Raw Time, Points Down, Final Time
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(12.dp)
                    .background(Color.White.copy(alpha = 0.05f), RoundedCornerShape(8.dp))
                    .padding(12.dp)
                    .clickable(onClick = onViewResult),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                MetricBox(
                    label = stringResource(R.string.idpa_raw_time),
                    value = formatTime(summary.totalTime),
                    modifier = Modifier.weight(1f)
                )

                MetricBox(
                    label = stringResource(R.string.idpa_points_down),
                    value = "${abs(pointsDown)}",
                    modifier = Modifier.weight(1f)
                )

                MetricBox(
                    label = stringResource(R.string.idpa_final_time),
                    value = formatTime(finalTime),
                    valueColor = Color.Red,
                    modifier = Modifier.weight(1f)
                )
            }

            // Zone breakdown
            IDPAZoneBreakdownView(breakdown = breakdown, pointsDown = pointsDown)

            // Replay button
            Button(
                onClick = onReplay,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(48.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.Red.copy(alpha = 0.3f),
                    contentColor = Color.White
                ),
                shape = RoundedCornerShape(8.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.PlayArrow,
                    contentDescription = null,
                    modifier = Modifier.size(18.dp)
                )
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = stringResource(R.string.drill_replay),
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 14.sp
                )
            }

            // Edit zones button
            Button(
                onClick = onEditZones,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(48.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = Color.Blue.copy(alpha = 0.3f),
                    contentColor = Color.White
                ),
                shape = RoundedCornerShape(8.dp)
            ) {
                Text(
                    text = stringResource(R.string.edit_zones),
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 14.sp
                )
            }
        }
    }
}

@Composable
private fun MetricBox(
    label: String,
    value: String,
    valueColor: Color = Color.White,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = label.uppercase(),
            style = MaterialTheme.typography.labelSmall,
            color = Color.White.copy(alpha = 0.7f),
            fontWeight = FontWeight.SemiBold,
            fontSize = 10.sp
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = value,
            style = MaterialTheme.typography.titleSmall,
            color = valueColor,
            fontWeight = FontWeight.Medium,
            textAlign = TextAlign.Center
        )
    }
}

@Composable
private fun IDPAZoneBreakdownView(
    breakdown: Map<String, Int>,
    pointsDown: Int,
    modifier: Modifier = Modifier
) {
    val zoneColors = mapOf(
        "Head" to Color(0xFF80FF80),    // Light green
        "Body" to Color(0xFFFFCC33),    // Yellow
        "Other" to Color(0xFFFF8033),   // Orange
        "Miss" to Color(0xFFFF5555)     // Red
    )

    val zonePoints = mapOf(
        "Head" to 0,
        "Body" to -1,
        "Other" to -3,
        "Miss" to -5
    )

    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(Color.White.copy(alpha = 0.03f))
            .border(1.dp, Color.White.copy(alpha = 0.1f), RoundedCornerShape(12.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text(
            text = stringResource(R.string.idpa_zones_breakdown),
            style = MaterialTheme.typography.labelSmall,
            color = Color.White.copy(alpha = 0.7f),
            fontWeight = FontWeight.SemiBold
        )

        // Zone badges
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            listOf("Head", "Body", "Other", "Miss").forEach { zone ->
                val count = breakdown[zone] ?: 0
                val points = zonePoints[zone] ?: 0
                val color = zoneColors[zone] ?: Color.Gray

                IDPAZoneBadge(
                    zone = zone,
                    count = count,
                    points = points,
                    color = color,
                    modifier = Modifier.weight(1f)
                )
            }
        }

        // Total points down
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = stringResource(R.string.idpa_total_points_down),
                style = MaterialTheme.typography.labelSmall,
                color = Color.White.copy(alpha = 0.7f),
                fontWeight = FontWeight.SemiBold
            )
            Text(
                text = "-${abs(pointsDown)}",
                style = MaterialTheme.typography.titleSmall,
                color = Color.Red,
                fontWeight = FontWeight.Bold
            )
        }
    }
}

@Composable
private fun IDPAZoneBadge(
    zone: String,
    count: Int,
    points: Int,
    color: Color,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .clip(RoundedCornerShape(12.dp))
            .background(color.copy(alpha = 0.1f))
            .border(2.dp, color, RoundedCornerShape(12.dp))
            .padding(12.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Box(
            modifier = Modifier
                .size(28.dp)
                .background(color, CircleShape),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = zone.take(1),
                color = Color.White,
                fontWeight = FontWeight.Bold,
                style = MaterialTheme.typography.labelMedium
            )
        }
        Spacer(modifier = Modifier.height(6.dp))
        Text(
            text = "$count",
            color = Color.White,
            fontWeight = FontWeight.SemiBold,
            style = MaterialTheme.typography.bodyLarge,
            textAlign = TextAlign.Center
        )
        Text(
            text = if (points == 0) "0" else "$points",
            color = if (points < 0) Color.Red else Color.Green,
            fontWeight = FontWeight.SemiBold,
            style = MaterialTheme.typography.labelSmall,
            fontSize = 10.sp
        )
    }
}

@Composable
private fun EmptyStateView() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.Black),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Text(
            text = stringResource(R.string.no_results_available),
            style = MaterialTheme.typography.headlineSmall,
            color = Color.White,
            fontWeight = FontWeight.Medium
        )
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            text = stringResource(R.string.complete_drill_message),
            style = MaterialTheme.typography.bodyMedium,
            color = Color.Gray,
            textAlign = TextAlign.Center
        )
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
private fun IDPAZoneEditDialog(
    summary: DrillRepeatSummary,
    onSave: (Map<String, Int>) -> Unit,
    onCancel: () -> Unit
) {
    val breakdown = ScoringUtility.getIDPAZoneBreakdown(summary.shots)

    var headCount by remember { mutableStateOf(breakdown["Head"] ?: 0) }
    var bodyCount by remember { mutableStateOf(breakdown["Body"] ?: 0) }
    var otherCount by remember { mutableStateOf(breakdown["Other"] ?: 0) }
    var missCount by remember { mutableStateOf(breakdown["Miss"] ?: 0) }

    Dialog(onDismissRequest = onCancel) {
        Card(
            modifier = Modifier
                .fillMaxWidth(0.85f)
                .padding(16.dp),
            shape = RoundedCornerShape(24.dp),
            colors = CardDefaults.cardColors(containerColor = Color.Black)
        ) {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(24.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(24.dp)
            ) {
                Text(
                    text = stringResource(R.string.edit_idpa_zones),
                    style = MaterialTheme.typography.headlineSmall,
                    color = Color.White,
                    fontWeight = FontWeight.Bold
                )

                // Zone editors
                ZoneEditorField(stringResource(R.string.idpa_zone_head), headCount) { headCount = it }
                ZoneEditorField(stringResource(R.string.idpa_zone_body), bodyCount) { bodyCount = it }
                ZoneEditorField(stringResource(R.string.idpa_zone_other), otherCount) { otherCount = it }
                ZoneEditorField(stringResource(R.string.idpa_zone_miss), missCount) { missCount = it }

                Row(
                    modifier = Modifier
                        .fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    OutlinedButton(
                        onClick = onCancel,
                        modifier = Modifier.weight(1f),
                        colors = ButtonDefaults.outlinedButtonColors(
                            contentColor = Color.White
                        )
                    ) {
                        Text(stringResource(R.string.cancel))
                    }
                    Button(
                        onClick = {
                            val updatedZones = mapOf(
                                "Head" to headCount,
                                "Body" to bodyCount,
                                "Other" to otherCount,
                                "Miss" to missCount
                            )
                            onSave(updatedZones)
                        },
                        modifier = Modifier.weight(1f),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = Color.Red
                        )
                    ) {
                        Text(stringResource(R.string.save))
                    }
                }
            }
        }
    }
}

@Composable
private fun ZoneEditorField(
    label: String,
    value: Int,
    onValueChange: (Int) -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(
            text = label,
            color = Color.White,
            style = MaterialTheme.typography.bodyLarge
        )

        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            IconButton(
                onClick = { if (value > 0) onValueChange(value - 1) },
                modifier = Modifier.size(32.dp)
            ) {
                Icon(
                    Icons.Default.Remove,
                    contentDescription = "Decrease",
                    tint = Color.Red
                )
            }

            Text(
                text = value.toString(),
                color = Color.White,
                style = MaterialTheme.typography.bodyLarge,
                modifier = Modifier.width(32.dp),
                textAlign = TextAlign.Center
            )

            IconButton(
                onClick = { onValueChange(value + 1) },
                modifier = Modifier.size(32.dp)
            ) {
                Icon(
                    Icons.Default.Add,
                    contentDescription = "Increase",
                    tint = Color.Blue
                )
            }
        }
    }
}

private fun formatTime(time: Double): String {
    return if (time.isFinite() && time > 0) {
        String.format("%.1f s", time)
    } else {
        "--"
    }
}

File: FlexTargetiOS/flextarget/View/Drills/IDPADrillSummaryView.swift
import SwiftUI
import CoreData

struct IDPADrillSummaryView: View {
    let drillSetup: DrillSetup
    @State var summaries: [DrillRepeatSummary]
    
    @Environment(\.dismiss) private var dismiss
    @Environment(\.managedObjectContext) private var environmentContext
    
    @State private var showEditDialog = false
    @State private var editingSummary: DrillRepeatSummary?
    @State private var showDrillResult = false
    @State private var showDrillReplay = false
    @State private var selectedSummary: DrillRepeatSummary?
    
    private var viewContext: NSManagedObjectContext {
        if let coordinator = environmentContext.persistentStoreCoordinator,
           coordinator.persistentStores.isEmpty == false {
            return environmentContext
        }
        return PersistenceController.shared.container.viewContext
    }
    
    private var drillName: String {
        drillSetup.name ?? "Untitled Drill"
    }
    
    var body: some View {
        NavigationView {
            ZStack {
                Color.black.ignoresSafeArea()
                
                VStack(spacing: 0) {
                    navigationBar
                    
                    if summaries.isEmpty {
                        emptyState
                    } else {
                        ScrollView(.vertical, showsIndicators: false) {
                            VStack(spacing: 24) {
                                ForEach(summaries.indices, id: \.self) { index in
                                    idpaDrillCard(for: summaries[index], index: index)
                                }
                            }
                            .padding(.vertical, 24)
                        }
                    }
                }
            }
            .navigationBarHidden(true)
            
            // Edit Dialog
            if showEditDialog, let editingSummary = editingSummary {
                IDPAZoneEditDialog(
                    summary: editingSummary,
                    onSave: { updatedZones in
                        if let index = summaries.firstIndex(where: { $0.id == editingSummary.id }) {
                            summaries[index].idpaZones = updatedZones
                        }
                        showEditDialog = false
                        self.editingSummary = nil
                    },
                    onCancel: {
                        showEditDialog = false
                        self.editingSummary = nil
                    }
                )
            }
            
            // Navigation to DrillResultView
            if showDrillResult, let selectedSummary = selectedSummary {
                NavigationLink(
                    destination: DrillResultView(drillSetup: drillSetup, repeatSummary: selectedSummary),
                    isActive: $showDrillResult
                ) {
                    EmptyView()
                }
            }
            
            // Navigation to DrillReplayView
            if showDrillReplay, let selectedSummary = selectedSummary {
                NavigationLink(
                    destination: DrillReplayView(drillSetup: drillSetup, shots: selectedSummary.shots),
                    isActive: $showDrillReplay
                ) {
                    EmptyView()
                }
            }
        }
    }
    
    // MARK: - Navigation Bar
    
    private var navigationBar: some View {
        HStack {
            Button(action: { dismiss() }) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(.white)
            }
            Spacer()
            Text(drillName.capitalized)
                .font(.system(size: 16, weight: .bold))
                .foregroundColor(.white)
            Spacer()
            // Placeholder for alignment
            Image(systemName: "chevron.left")
                .font(.system(size: 16, weight: .semibold))
                .foregroundColor(.white)
                .opacity(0)
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 16)
        .background(Color.black.opacity(0.6))
    }
    
    // MARK: - IDPA Drill Card
    
    private func idpaDrillCard(for summary: DrillRepeatSummary, index: Int) -> some View {
        let breakdown = ScoringUtility.getIDPAZoneBreakdown(shots: summary.shots)
        let pointsDown = ScoringUtility.calculateIDPAPointsDown(shots: summary.shots)
        let finalTime = ScoringUtility.calculateIDPAFinalTime(rawTime: summary.totalTime, pointsDown: pointsDown)
        
        return VStack(spacing: 16) {
            // Header: Repeat number and basic stats
            VStack(spacing: 12) {
                HStack {
                    Text(String(format: NSLocalizedString("repeat_number", comment: "Repeat number format"), summary.repeatIndex))
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                    Spacer()
                }
                
                // Metrics grid: Raw Time, Final Time (Score) - CLICKABLE
                Button(action: {
                    selectedSummary = summary
                    showDrillResult = true
                }) {
                    HStack(spacing: 16) {
                        VStack(spacing: 4) {
                            Text(NSLocalizedString("idpa_raw_time", comment: "Raw time label"))
                                .font(.system(size: 11, weight: .semibold))
                                .foregroundColor(.white.opacity(0.7))
                            Text(format(time: summary.totalTime))
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.white)
                        }
                        .frame(minWidth: 80)
                        
                        VStack(spacing: 4) {
                            Text(NSLocalizedString("idpa_points_down", comment: "Points down label"))
                                .font(.system(size: 11, weight: .semibold))
                                .foregroundColor(.white.opacity(0.7))
                            Text("\(abs(pointsDown))")
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.white)
                        }
                        .frame(minWidth: 80)

                        Spacer()
                        
                        VStack(spacing: 4) {
                            Text(NSLocalizedString("idpa_final_time", comment: "Final time label"))
                                .font(.system(size: 11, weight: .semibold))
                                .foregroundColor(.white.opacity(0.7))
                            Text(format(time: finalTime))
                                .font(.system(size: 18, weight: .bold))
                                .foregroundColor(.red)
                        }
                        .frame(minWidth: 80)
                    }
                    .padding(.horizontal, 16)
                    .padding(.vertical, 12)
                    .frame(maxWidth: .infinity)
                    .background(Color.white.opacity(0.05))
                    .cornerRadius(8)
                }
            }
            .padding(16)
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.white.opacity(0.03))
                    .overlay(
                        RoundedRectangle(cornerRadius: 12)
                            .stroke(Color.white.opacity(0.1), lineWidth: 1)
                    )
            )
            .padding(.horizontal, 20)
            
            // Zone breakdown
            IDPAZoneBreakdownView(breakdown: breakdown, pointsDown: pointsDown)
                .padding(.horizontal, 20)
            
            // Replay button
            Button(action: {
                selectedSummary = summary
                showDrillReplay = true
            }) {
                HStack {
                    Image(systemName: "play.circle.fill")
                        .font(.system(size: 16, weight: .semibold))
                    Text(NSLocalizedString("drill_replay", comment: "Replay button"))
                        .font(.system(size: 14, weight: .semibold))
                }
                .foregroundColor(.white)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(Color.red.opacity(0.3))
                .cornerRadius(8)
            }
            .padding(.horizontal, 20)
            
            // Edit zones button
            Button(action: {
                editingSummary = summary
                showEditDialog = true
            }) {
                HStack {
                    Image(systemName: "pencil.circle.fill")
                        .font(.system(size: 16, weight: .semibold))
                    Text(NSLocalizedString("edit_zones", comment: "Edit zones button"))
                        .font(.system(size: 14, weight: .semibold))
                }
                .foregroundColor(.white)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 12)
                .background(Color.blue.opacity(0.3))
                .cornerRadius(8)
            }
            .padding(.horizontal, 20)
            .padding(.bottom, 8)
        }
        .padding(.vertical, 8)
    }
    
    // MARK: - Empty State
    
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.white.opacity(0.5))
            Text(NSLocalizedString("no_results_title", comment: "No results title"))
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            Text(NSLocalizedString("no_results_message", comment: "No results message"))
                .font(.system(size: 14, weight: .regular))
                .foregroundColor(.white.opacity(0.6))
                .multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
    
    // MARK: - Helpers
    
    private func format(time: TimeInterval) -> String {
        let minutes = Int(time) / 60
        let seconds = Int(time) % 60
        let milliseconds = Int((time.truncatingRemainder(dividingBy: 1)) * 1000)
        
        if minutes > 0 {
            return String(format: "%d:%02d.%03d", minutes, seconds, milliseconds)
        } else {
            return String(format: "%.3fs", time)
        }
    }
}

// MARK: - IDPA Zone Breakdown View

struct IDPAZoneBreakdownView: View {
    let breakdown: [String: Int]
    let pointsDown: Int
    
    private let zoneColors: [String: Color] = [
        "Head": Color(red: 0.5, green: 1.0, blue: 0.5),  // Light green
        "Body": Color(red: 1.0, green: 0.8, blue: 0.2),  // Yellow
        "Other": Color(red: 1.0, green: 0.5, blue: 0.2), // Orange
        "Miss": Color(red: 1.0, green: 0.3, blue: 0.3)   // Red
    ]
    
    private let zonePoints: [String: Int] = [
        "Head": 0,
        "Body": -1,
        "Other": -3,
        "Miss": -5
    ]
    
    var body: some View {
        VStack(spacing: 12) {
            Text(NSLocalizedString("idpa_zones_breakdown", comment: "Zones breakdown title"))
                .font(.system(size: 12, weight: .semibold))
                .foregroundColor(.white.opacity(0.7))
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.horizontal, 4)
            
            HStack(spacing: 8) {
                ForEach(["Head", "Body", "Other", "Miss"], id: \.self) { zone in
                    let count = breakdown[zone] ?? 0
                    let points = zonePoints[zone] ?? 0
                    
                    VStack(spacing: 6) {
                        Circle()
                            .fill(zoneColors[zone] ?? .gray)
                            .frame(width: 28, height: 28)
                            .overlay(
                                Text(zone.prefix(1))
                                    .font(.system(size: 11, weight: .bold))
                                    .foregroundColor(.black)
                            )
                        
                        Text("\(count)")
                            .font(.system(size: 14, weight: .bold))
                            .foregroundColor(.white)
                        
                        if points != 0 {
                            Text("\(points)")
                                .font(.system(size: 10, weight: .semibold))
                                .foregroundColor(points < 0 ? Color.red : Color.green)
                        } else {
                            Text("0")
                                .font(.system(size: 10, weight: .semibold))
                                .foregroundColor(.white.opacity(0.5))
                        }
                    }
                    .frame(maxWidth: .infinity)
                    .padding(.vertical, 12)
                    .background(zoneColors[zone]?.opacity(0.1) ?? Color.gray.opacity(0.1))
                    .cornerRadius(8)
                }
            }
            
            // Total points down summary
            HStack {
                Text(NSLocalizedString("idpa_total_points_down", comment: "Total points down label"))
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundColor(.white.opacity(0.7))
                Spacer()
                Text(String(format: "-%d", abs(pointsDown)))
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.red)
            }
            .padding(.horizontal, 4)
            .padding(.top, 4)
        }
        .padding(16)
        .background(
            RoundedRectangle(cornerRadius: 12)
                .fill(Color.white.opacity(0.03))
                .overlay(
                    RoundedRectangle(cornerRadius: 12)
                        .stroke(Color.white.opacity(0.1), lineWidth: 1)
                )
        )
    }
}

// MARK: - Edit Dialog

struct IDPAZoneEditDialog: View {
    let summary: DrillRepeatSummary
    let onSave: ([String: Int]) -> Void
    let onCancel: () -> Void
    
    @State private var headCount: Int
    @State private var bodyCount: Int
    @State private var otherCount: Int
    @State private var missCount: Int
    
    init(summary: DrillRepeatSummary, onSave: @escaping ([String: Int]) -> Void, onCancel: @escaping () -> Void) {
        self.summary = summary
        self.onSave = onSave
        self.onCancel = onCancel
        
        let breakdown = ScoringUtility.getIDPAZoneBreakdown(shots: summary.shots)
        _headCount = State(initialValue: breakdown["Head"] ?? 0)
        _bodyCount = State(initialValue: breakdown["Body"] ?? 0)
        _otherCount = State(initialValue: breakdown["Other"] ?? 0)
        _missCount = State(initialValue: breakdown["Miss"] ?? 0)
    }
    
    var body: some View {
        ZStack {
            Color.black.opacity(0.5).ignoresSafeArea()
            
            VStack(spacing: 20) {
                Text(NSLocalizedString("edit_idpa_zones", comment: "Edit IDPA zones title"))
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(.white)
                
                VStack(spacing: 16) {
                    zoneEditor(label: NSLocalizedString("idpa_zone_head", comment: "Head zone"), count: $headCount)
                    zoneEditor(label: NSLocalizedString("idpa_zone_body", comment: "Body zone"), count: $bodyCount)
                    zoneEditor(label: NSLocalizedString("idpa_zone_other", comment: "Other zone"), count: $otherCount)
                    zoneEditor(label: NSLocalizedString("idpa_zone_miss", comment: "Miss zone"), count: $missCount)
                }
                
                HStack(spacing: 12) {
                    Button(action: onCancel) {
                        Text(NSLocalizedString("cancel", comment: "Cancel button"))
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 12)
                            .background(Color.white.opacity(0.1))
                            .cornerRadius(8)
                    }
                    
                    Button(action: {
                        let updatedZones: [String: Int] = [
                            "Head": headCount,
                            "Body": bodyCount,
                            "Other": otherCount,
                            "Miss": missCount
                        ]
                        onSave(updatedZones)
                    }) {
                        Text(NSLocalizedString("save", comment: "Save button"))
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 12)
                            .background(Color.blue)
                            .cornerRadius(8)
                    }
                }
                
                Spacer()
            }
            .padding(24)
            .background(
                RoundedRectangle(cornerRadius: 16)
                    .fill(Color(red: 0.1, green: 0.1, blue: 0.1))
            )
            .padding(20)
        }
    }
    
    private func zoneEditor(label: String, count: Binding<Int>) -> some View {
        HStack {
            Text(label)
                .font(.system(size: 14, weight: .semibold))
                .foregroundColor(.white)
            
            Spacer()
            
            HStack(spacing: 12) {
                Button(action: { if count.wrappedValue > 0 { count.wrappedValue -= 1 } }) {
                    Image(systemName: "minus.circle.fill")
                        .font(.system(size: 20))
                        .foregroundColor(.red)
                }
                
                Text("\(count.wrappedValue)")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(minWidth: 30)
                
                Button(action: { count.wrappedValue += 1 }) {
                    Image(systemName: "plus.circle.fill")
                        .font(.system(size: 20))
                        .foregroundColor(.blue)
                }
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color.white.opacity(0.05))
        .cornerRadius(8)
    }
}

#Preview {
    let mockSummary = DrillRepeatSummary(
        repeatIndex: 1,
        totalTime: 25.5,
        numShots: 4,
        firstShot: 0.5,
        fastest: 0.8,
        score: 0,
        shots: []
    )
    
    let mockDrillSetup = DrillSetup(context: PersistenceController.preview.container.viewContext)
    mockDrillSetup.name = "IDPA Drill"
    mockDrillSetup.mode = "idpa"
    
    return NavigationStack {
        IDPADrillSummaryView(drillSetup: mockDrillSetup, summaries: [mockSummary])
    }
}
